elf/input-files.cc

```diff
@@ -1535,6 +1535,7 @@ template <typename E>
 void SharedFile<E>::populate_symtab(Context<E> &ctx) {
   ElfSym<E> *symtab =
     (ElfSym<E> *)(ctx.buf + ctx.symtab->shdr.sh_offset) + this->global_symtab_idx;
+  memset(symtab, 0, sizeof(ElfSym<E>) * this->num_global_symtab);
```

Comment:
Not a big deal but I think it is slightly better if we move this memset inside the loop and clear each ElfSym individually just before setting values to the struct members. I believe compiler is smart enough to fuse the memset and struct member assignments. Also it should improve memory locality.